ledfx/utilities/gradient_extraction.py

```
@@ -0,0 +1,986 @@
+                    "hsv": [h, s, v],
+                    "frequency": frequency,
+                }
+            )
+
+        # Sort by frequency (most dominant first)
+        color_frequencies.sort(key=lambda x: x["frequency"], reverse=True)
+
+        # Deduplicate similar colors to prevent gradient dominated by close variants
+        deduplicated = _deduplicate_colors(color_frequencies)
+
+        return deduplicated
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)